### DISCUSSION TITLE
emails "You have free access to Github Copilot" legit? Is Github using sendgrid.net in its emails?

### DISCUSSION THREAD

**Discussion by thediveo** [community/community#147438] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

```
Received: from xtrwsrct.outbound-mail.sendgrid.net ([167.89.101.199])
From: GitHub <no-reply@github.com>
Subject: You have free access to GitHub Copilot
```

It contains HTML parts with a button reading "Start using Copilot" and linking to `https://u15762298.ct.sendgrid.net/ls/click?...`

The plain text part has an unsubscribe link to sendgrid.net, but the other links are directly linking to github.com.

Is this a Github genuine emailing?

**Comment by samus-aran** [comment]
Hi @thediveo 👋 

We did send out emails to users stating we've launched GitHub Copilot Free. Check out our [blog post on this announcement](https://github.blog/news-insights/product-news/github-copilot-in-vscode-free/). Sorry for any confusion.

**Reply by thediveo** [reply]
No problem at all, it is just that I want to make sure that no-one is actually copying your email, but with nasty links put in between. So Github is using u15762298.ct.sendgrid.net links?

**Reply by materia79** [reply]
Why can't they just use a github.com subdomain? Crazy..

**Comment by uportalis** [comment]
Hi,
The URLs are blocked by Microsoft's EDR.
![image](https://github.com/user-attachments/assets/775046ef-0528-4225-bc48-428eefa38609)

**Reply by thediveo** [reply]
are there more details?

**Comment by thediveo** [comment]
I've opened a support ticket in the hope to get an official response, and if necessary, make Github aware of the situation.

**Reply by fcosta-alb** [reply]
Same here: the email looks legit, however the URLs are blocked by windows.
![image](https://github.com/user-attachments/assets/03fb575b-3c4b-4a94-8975-6c91b3999343)

**Comment by ohjohnsen** [comment]
Just received this email as well. Firefox is blocking the link, reporting it as a possible phishing email.

<img width="1026" alt="bilde" src="https://github.com/user-attachments/assets/1a964842-f268-4c9e-b03c-db1513ed754e" />

I think GitHub blundered here, using links with domains other than github.com. This is always a red flag for me, causing me actually to search for and find this discussion.

**Comment by samus-aran** [comment, marked as answer]
Hi y'all 👋 

Thank you so much for your feedback and for providing the helpful screenshots! We truly appreciate the time you’ve taken to bring these concerns to our attention.

We can confirm that we do indeed utilize sendgrid as part of our communication process, so these emails are indeed legitimate. The links embedded within the emails are platform redirect links used primarily for click tracking purposes.

However, we understand your concerns and will certainly take your feedback into account, forwarding it to our internal teams to address any potential issues. Thank you for raising this with us.

**Reply by thediveo** [reply]
Thank you very much! In view of potential supply chain attack vectors, I hope that the internal teams hopefully will take into account not to sabotage the good Github standing by falling back to silky marketing behavior.
